File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/mailFolders/MailFolder.java
package graphjavav4.utilities.users.mailFolders;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import graphjavav4.utilities.users.Entity;
import graphjavav4.utilities.users.Message;
import graphjavav4.utilities.users.MultiValueLegacyExtendedProperty;
import graphjavav4.utilities.users.SingleValueLegacyExtendedProperty;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Objects;
public class MailFolder extends Entity implements Parsable {
    /** The number of immediate child mailFolders in the current mailFolder.  */
    @javax.annotation.Nullable
    public Integer childFolderCount;
    /** The mailFolder's display name.  */
    @javax.annotation.Nullable
    public String displayName;
    /** The unique identifier for the mailFolder's parent mailFolder.  */
    @javax.annotation.Nullable
    public String parentFolderId;
    /** The number of items in the mailFolder.  */
    @javax.annotation.Nullable
    public Integer totalItemCount;
    /** The number of items in the mailFolder marked as unread.  */
    @javax.annotation.Nullable
    public Integer unreadItemCount;
    /** The collection of child folders in the mailFolder.  */
    @javax.annotation.Nullable
    public List<MailFolder> childFolders;
    /** The collection of rules that apply to the user's Inbox folder.  */
    @javax.annotation.Nullable
    public List<MessageRule> messageRules;
    /** The collection of messages in the mailFolder.  */
    @javax.annotation.Nullable
    public List<Message> messages;
    /** The collection of multi-value extended properties defined for the mailFolder. Read-only. Nullable.  */
    @javax.annotation.Nullable
    public List<MultiValueLegacyExtendedProperty> multiValueExtendedProperties;
    /** The collection of single-value extended properties defined for the mailFolder. Read-only. Nullable.  */
    @javax.annotation.Nullable
    public List<SingleValueLegacyExtendedProperty> singleValueExtendedProperties;
    /**
     * Serialiazes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @return a void
     */
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        super.serialize(writer);
        writer.writeIntegerValue("childFolderCount", childFolderCount);
        writer.writeStringValue("displayName", displayName);
        writer.writeStringValue("parentFolderId", parentFolderId);
        writer.writeIntegerValue("totalItemCount", totalItemCount);
        writer.writeIntegerValue("unreadItemCount", unreadItemCount);
        writer.writeCollectionOfObjectValues("childFolders", childFolders);
        writer.writeCollectionOfObjectValues("messageRules", messageRules);
        writer.writeCollectionOfObjectValues("messages", messages);
        writer.writeCollectionOfObjectValues("multiValueExtendedProperties", multiValueExtendedProperties);
        writer.writeCollectionOfObjectValues("singleValueExtendedProperties", singleValueExtendedProperties);
    }
    /**
     * The serialization information for the current model
     * @return a Map<String, BiConsumer<T, ParseNode>>
     */
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getDeserializeFields() {
        final Map<String, BiConsumer<T, ParseNode>> fields = new HashMap<>(super.getDeserializeFields());
        fields.put("childFolderCount", (o, n) -> { ((MailFolder)o).childFolderCount = n.getIntegerValue(); });
        fields.put("displayName", (o, n) -> { ((MailFolder)o).displayName = n.getStringValue(); });
        fields.put("parentFolderId", (o, n) -> { ((MailFolder)o).parentFolderId = n.getStringValue(); });
        fields.put("totalItemCount", (o, n) -> { ((MailFolder)o).totalItemCount = n.getIntegerValue(); });
        fields.put("unreadItemCount", (o, n) -> { ((MailFolder)o).unreadItemCount = n.getIntegerValue(); });
        fields.put("childFolders", (o, n) -> { ((MailFolder)o).childFolders = n.getCollectionOfObjectValues(MailFolder.class); });
        fields.put("messageRules", (o, n) -> { ((MailFolder)o).messageRules = n.getCollectionOfObjectValues(MessageRule.class); });
        fields.put("messages", (o, n) -> { ((MailFolder)o).messages = n.getCollectionOfObjectValues(Message.class); });
        fields.put("multiValueExtendedProperties", (o, n) -> { ((MailFolder)o).multiValueExtendedProperties = n.getCollectionOfObjectValues(MultiValueLegacyExtendedProperty.class); });
        fields.put("singleValueExtendedProperties", (o, n) -> { ((MailFolder)o).singleValueExtendedProperties = n.getCollectionOfObjectValues(SingleValueLegacyExtendedProperty.class); });
        return fields;
    }
}

File: msgraph-mail/typescript/src/models/mailFolder.ts
import { deserializeIntoEntity, serializeEntity, type Entity } from './entity';
import { createMessageFromDiscriminatorValue, serializeMessage, type Message } from './message';
import { createMessageRuleFromDiscriminatorValue, serializeMessageRule, type MessageRule } from './messageRule';
import { createMultiValueLegacyExtendedPropertyFromDiscriminatorValue, serializeMultiValueLegacyExtendedProperty, type MultiValueLegacyExtendedProperty } from './multiValueLegacyExtendedProperty';
import { createSingleValueLegacyExtendedPropertyFromDiscriminatorValue, serializeSingleValueLegacyExtendedProperty, type SingleValueLegacyExtendedProperty } from './singleValueLegacyExtendedProperty';
import { type Parsable, type ParseNode, type SerializationWriter } from '@microsoft/kiota-abstractions';

export function createMailFolderFromDiscriminatorValue(parseNode: ParseNode | undefined) {
    if(!parseNode) throw new Error("parseNode cannot be undefined");
    return deserializeIntoMailFolder;
}
export function deserializeIntoMailFolder(mailFolder: MailFolder | undefined = {} as MailFolder) : Record<string, (node: ParseNode) => void> {
    return {
        ...deserializeIntoEntity(mailFolder),
        "childFolderCount": n => { mailFolder.childFolderCount = n.getNumberValue(); },
        "childFolders": n => { mailFolder.childFolders = n.getCollectionOfObjectValues<MailFolder>(createMailFolderFromDiscriminatorValue); },
        "displayName": n => { mailFolder.displayName = n.getStringValue(); },
        "isHidden": n => { mailFolder.isHidden = n.getBooleanValue(); },
        "messageRules": n => { mailFolder.messageRules = n.getCollectionOfObjectValues<MessageRule>(createMessageRuleFromDiscriminatorValue); },
        "messages": n => { mailFolder.messages = n.getCollectionOfObjectValues<Message>(createMessageFromDiscriminatorValue); },
        "multiValueExtendedProperties": n => { mailFolder.multiValueExtendedProperties = n.getCollectionOfObjectValues<MultiValueLegacyExtendedProperty>(createMultiValueLegacyExtendedPropertyFromDiscriminatorValue); },
        "parentFolderId": n => { mailFolder.parentFolderId = n.getStringValue(); },
        "singleValueExtendedProperties": n => { mailFolder.singleValueExtendedProperties = n.getCollectionOfObjectValues<SingleValueLegacyExtendedProperty>(createSingleValueLegacyExtendedPropertyFromDiscriminatorValue); },
        "totalItemCount": n => { mailFolder.totalItemCount = n.getNumberValue(); },
        "unreadItemCount": n => { mailFolder.unreadItemCount = n.getNumberValue(); },
    }
}
export interface MailFolder extends Entity, Parsable {
    /**
     * The number of immediate child mailFolders in the current mailFolder.
     */
    childFolderCount?: number | undefined;
    /**
     * The collection of child folders in the mailFolder.
     */
    childFolders?: MailFolder[] | undefined;
    /**
     * The mailFolder's display name.
     */
    displayName?: string | undefined;
    /**
     * Indicates whether the mailFolder is hidden. This property can be set only when creating the folder. Find more information in Hidden mail folders.
     */
    isHidden?: boolean | undefined;
    /**
     * The collection of rules that apply to the user's Inbox folder.
     */
    messageRules?: MessageRule[] | undefined;
    /**
     * The collection of messages in the mailFolder.
     */
    messages?: Message[] | undefined;
    /**
     * The collection of multi-value extended properties defined for the mailFolder. Read-only. Nullable.
     */
    multiValueExtendedProperties?: MultiValueLegacyExtendedProperty[] | undefined;
    /**
     * The unique identifier for the mailFolder's parent mailFolder.
     */
    parentFolderId?: string | undefined;
    /**
     * The collection of single-value extended properties defined for the mailFolder. Read-only. Nullable.
     */
    singleValueExtendedProperties?: SingleValueLegacyExtendedProperty[] | undefined;
    /**
     * The number of items in the mailFolder.
     */
    totalItemCount?: number | undefined;
    /**
     * The number of items in the mailFolder marked as unread.
     */
    unreadItemCount?: number | undefined;
}
export function serializeMailFolder(writer: SerializationWriter, mailFolder: MailFolder | undefined = {} as MailFolder) : void {
        serializeEntity(writer, mailFolder)
        writer.writeNumberValue("childFolderCount", mailFolder.childFolderCount);
        writer.writeCollectionOfObjectValues<MailFolder>("childFolders", mailFolder.childFolders, );
        writer.writeStringValue("displayName", mailFolder.displayName);
        writer.writeBooleanValue("isHidden", mailFolder.isHidden);
        writer.writeCollectionOfObjectValues<MessageRule>("messageRules", mailFolder.messageRules, );
        writer.writeCollectionOfObjectValues<Message>("messages", mailFolder.messages, );
        writer.writeCollectionOfObjectValues<MultiValueLegacyExtendedProperty>("multiValueExtendedProperties", mailFolder.multiValueExtendedProperties, );
        writer.writeStringValue("parentFolderId", mailFolder.parentFolderId);
        writer.writeCollectionOfObjectValues<SingleValueLegacyExtendedProperty>("singleValueExtendedProperties", mailFolder.singleValueExtendedProperties, );
        writer.writeNumberValue("totalItemCount", mailFolder.totalItemCount);
        writer.writeNumberValue("unreadItemCount", mailFolder.unreadItemCount);
}
